### PR TITLE
refactor get-oncall and use terraform to deploy

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
refactor to use cloud run and terraform, this to prepare to another service to come (the release notes bot) :) 

and with this will be easier to add other services in case we need and maintain

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    # Attempt to deploy the terraform configuration
+    - uses: hashicorp/setup-terraform@651471c36a6092792c552e8b1bef71e592b462d8 # v2.0.0
+      with:
+        terraform_version: 1.8
+
+    - working-directory: ./iac
+      run: |
+        terraform init
+
+        terraform plan
```
Comment (bobcallaway) [inline .github/workflows/deploy.yaml:46]: should this write the plan to a temporary file and then use that file in the apply command below? there is a race condition possible here.

Comment (cpanato) [reply .github/workflows/deploy.yaml:46]: yep

**Comment by cpanato** [comment]
ping @bobcallaway

File: .github/workflows/terraform.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+jobs:
+  lint-and-validate:
+    name: terraform-lint-validate
+    runs-on: ubuntu-latest
+
+    if: github.repository == 'sigstore/sigstore-devops-tools'
+
+    steps:
+      - uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+      - uses: hashicorp/setup-terraform@651471c36a6092792c552e8b1bef71e592b462d8 # v3.1.1
```
Comment (bobcallaway) [inline .github/workflows/terraform.yaml:21]: should we use the same version as in the deploy step?

File: iac/get_oncall.tf [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+resource "google_secret_manager_secret_iam_binding" "slack_api_secret_binding" {
+  secret_id = data.google_secret_manager_secret.slack_secret.id
+  role      = "roles/secretmanager.secretAccessor"
+  members = [
+    "serviceAccount:${google_service_account.slack_slash_pg_sa.email}",
+  ]
+}
+
+module "slack_slash_pg_service" {
+  source  = "chainguard-dev/common/infra//modules/regional-go-service"
```
Comment (bobcallaway) [inline iac/get_oncall.tf:25]: will this work? I doubt that source code is public?

File: iac/main.tf [hunk 4]
```diff
@@ -0,0 +1,57 @@
+# Copyright 2024 The Sigstore Authors
+# SPDX-License-Identifier: Apache-2.0
+
+provider "google" { project = var.project_id }
+provider "google-beta" { project = var.project_id }
+provider "ko" { docker_repo = "gcr.io/${var.project_id}" }
+
+// Create a network with several regional subnets
+module "networking" {
+  source  = "chainguard-dev/common/infra//modules/networking"
```
Comment (bobcallaway) [inline iac/main.tf:10]: same here - can we use OSS for this?

File: iac/get_oncall.tf (hunk 3 above)
Comment (cpanato) [reply iac/get_oncall.tf:25]: it is :) : https://github.com/chainguard-dev/terraform-infra-common

it a working that we have being doing to encapsulate and make it easier to use

File: iac/main.tf (hunk 4 above)
Comment (cpanato) [reply iac/main.tf:10]: it is oss :) https://github.com/chainguard-dev/terraform-infra-common

Comment (cpanato) [reply iac/main.tf:10]: @bobcallaway if you don't like i can redo manually

Comment (bobcallaway) [reply iac/main.tf:10]: ok... must have been my jetlag that I didn't link it to being a terraform provider ... nm then :)

Comment (cpanato) [reply iac/main.tf:10]: https://registry.terraform.io/modules/chainguard-dev/common/infra/latest

File: .github/workflows/terraform.yaml (hunk 2 above)
Comment (bobcallaway) [reply .github/workflows/terraform.yaml:21]: just waiting on this to LGTM

Comment (cpanato) [reply .github/workflows/terraform.yaml:21]: ahh sorry
missed that

Comment (cpanato) [reply .github/workflows/terraform.yaml:21]: done

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**